oauth 1.2.4 broken client-side: Cannot find module 'url'

When the `oauth` package was upgraded to version 1.2.4, I see this error in the client-side JS console when starting my app:

    Uncaught Error: Cannot find module 'url'
        at makeMissingError (modules-runtime.js?hash=b819d45cbf32aff410d740fac0364cb4088cd3f2:232)
        at Module.require (modules-runtime.js?hash=b819d45cbf32aff410d740fac0364cb4088cd3f2:251)
        at Module.moduleLink [as link] (modules.js?hash=577e55e5372d19f0c986c19655297cc3462f0205:334)
        at oauth_common.js (oauth_common.js:1)
        at fileEvaluate (modules-runtime.js?hash=b819d45cbf32aff410d740fac0364cb4088cd3f2:346)
        at Module.require (modules-runtime.js?hash=b819d45cbf32aff410d740fac0364cb4088cd3f2:248)
        at require (modules-runtime.js?hash=b819d45cbf32aff410d740fac0364cb4088cd3f2:268)
        at oauth.js?hash=2b2db59bfd63458cce3c5d570cb2fbd8f786f135:365
        at oauth.js?hash=2b2db59bfd63458cce3c5d570cb2fbd8f786f135:374

Reverting `oauth` to 1.2.3 (by editing `.meteor/versions`) fixes the problem. Upgrading to 1.2.4 again brings the problem back.

This looks like a side-effect of tooling changes. The source code changes between 1.2.3 and 1.2.4 don't look in any way related to the observed problem. But, the diff of the post-build package is quite large. In particular it appears that the 1.2.3 build transpiled ES6 to ES5 whereas 1.2.4 does not. The following parts of the diff look relevant:

```diff
--- /home/kenton/.meteor/packages/oauth/1.2.3/os/oauth_common.js        2018-04-07 14:29:11.874502103 -0700
+++ /home/kenton/.meteor/packages/oauth/1.2.4/os/oauth_common.js        2018-10-20 09:32:54.387547067 -0700
@@ -1,6 +1,8 @@
+import url from 'url';
+
 OAuth._storageTokenPrefix = "Meteor.oauth.credentialSecret-";
 
-OAuth._redirectUri = function (serviceName, config, params, absoluteUrlOptions) {
+OAuth._redirectUri = (serviceName, config, params, absoluteUrlOptions) => {
   // XXX COMPAT WITH 0.9.0
   // The redirect URI used to have a "?close" query argument.  We
   // detect whether we need to be backwards compatible by checking for
@@ -36,8 +38,7 @@
       // XXX Maybe we should put this in a separate package or something
       // that is used here and by boilerplate-generator? Or maybe
       // `Meteor.absoluteUrl` should know how to do this?
-      var url = Npm.require("url");
-      var parsedRootUrl = url.parse(rootUrl);
+      const parsedRootUrl = url.parse(rootUrl);
       if (parsedRootUrl.hostname === "localhost") {
         parsedRootUrl.hostname = "10.0.2.2";
         delete parsedRootUrl.host;
```

It would appear that in 1.2.3, the `import url` line was transpiled to `Npm.require("url")` *and relocated to the scope where it is used* (which incidentally is inside an `if (Meteor.isServer)` block). But in 1.2.4, the `import url` line is left at the top level. It would appear that the `url` package is only available on the server and so the import fails on the client side -- but, previously, the transpiler had relocated the import so that it never happened client-side.

--- a/packages/oauth/oauth_common.js
+++ b/packages/oauth/oauth_common.js
@@ -1,5 +1,3 @@
-import url from 'url';
-
 OAuth._storageTokenPrefix = "Meteor.oauth.credentialSecret-";
 
 OAuth._redirectUri = (serviceName, config, params, absoluteUrlOptions) => {
@@ -29,6 +27,7 @@ OAuth._redirectUri = (serviceName, config, params, absoluteUrlOptions) => {
   }
 
   if (Meteor.isServer && isCordova) {
+    const url = Npm.require('url');
     let rootUrl = process.env.MOBILE_ROOT_URL ||
           __meteor_runtime_config__.ROOT_URL;
 
@@ -46,7 +45,7 @@ OAuth._redirectUri = (serviceName, config, params, absoluteUrlOptions) => {
       rootUrl = url.format(parsedRootUrl);
     }
 
-    absoluteUrlOptions = { 
+    absoluteUrlOptions = {
       ...absoluteUrlOptions,
       // For Cordova clients, redirect to the special Cordova root url
       // (likely a local IP in development mode).

--- a/packages/oauth/oauth_server.js
+++ b/packages/oauth/oauth_server.js
@@ -435,7 +435,7 @@ OAuth.sealSecret = plaintext => {
     return OAuthEncryption.seal(plaintext);
   else
     return plaintext;
-}
+};
 
 // Unencrypt a service data field, if the "oauth-encryption"
 // package is loaded and the field is encrypted.

--- a/packages/oauth/package.js
+++ b/packages/oauth/package.js
@@ -1,6 +1,6 @@
 Package.describe({
   summary: "Common code for OAuth-based services",
-  version: "1.2.4"
+  version: "1.2.5"
 });
 
 Package.onUse(api => {

